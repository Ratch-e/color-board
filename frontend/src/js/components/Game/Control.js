import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Control = ({color, click}) => (
	<div
		className={classNames('controls__item', `controls__item_color_${color}`)}
		onClick={() => click(color)}
	/>
);

Control.propTypes = {
	color: PropTypes.string,
	click: PropTypes.func,
};

export default Control;