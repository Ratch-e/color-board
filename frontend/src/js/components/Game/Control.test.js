import React from 'react';
import {shallow} from 'enzyme';
import Control from './Control';

describe('Элемент управления', () => {
	it('рендерится корректно', () => {
		const component = shallow(<Control/>);
		expect(component.length).toMatchSnapshot();
	});

	it('выбирает нужный класс в зависимости от переданного цвета', () => {
		const component = shallow(<Control color="red"/>);
		expect(component.find('.controls__item').hasClass('controls__item_color_red')).toEqual(true);
	});
});
