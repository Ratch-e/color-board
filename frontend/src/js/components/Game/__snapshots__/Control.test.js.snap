// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Элемент управления рендерится корректно 1`] = `1`;
